vm/src/pyobject.rs

```diff
@@ -643,8 +643,13 @@ impl PyContext {
         objtype::new(self.type_type(), name, vec![base], PyAttributes::new()).unwrap()
     }
 
-    pub fn new_scope(&self) -> Scope {
-        Scope::new(None, self.new_dict())
+    pub fn new_scope(&self, vm: &VirtualMachine) -> Scope {
+        let scope = Scope::new(None, self.new_dict());
+
+        let globals = scope.globals.clone();
+        let _ = globals.set_item("__builtins__", vm.builtins.clone(), vm);
```

Comment:
You can remove the `let _ =` part here. Underscore is the unused value. In case set_item returns an error, it should be handled by the `?` operator, or by unwrap, such that we panic when this fails. My preference is to handle the error with the `?` operator.